File: Roman to integer.py
class Solution:
    def romanToInt(self, s):
        result = 0
        roman_values = {
            'I': 1,
            'V': 5,
            'X': 10,
            'L': 50,
            'C': 100,
            'D': 500,
            'M': 1000
        }
        for i in range(len(s) - 1):
            current_value = roman_values[s[i]]
            next_value = roman_values[s[i + 1]]
            if current_value < next_value:
                result -= current_value
            else:
                result += current_value

        result += roman_values[s[-1]]
        return result

File: Roman To Integer.java
class Solution {
    public int romanToInt(String s) {
        Map<Character,Integer> map = new HashMap<>();
        map.put('I',1);
        map.put('V',5);
        map.put('X',10);
        map.put('L',50);
        map.put('C',100);
        map.put('D',500);
        map.put('M',1000);
        int result=0;
        int i,j;
        char Chars[]= s.toCharArray();
        for(i=0,j=1;j<Chars.length;i++,j++){
            if(map.get(Chars[i])>=map.get(Chars[j])){
                result+=map.get(Chars[i]);
            }else{
                result-=map.get(Chars[i]);
            }
        }
        result+=map.get(Chars[i]);
        return result;
    }
}
